data/src/test/java/org/apache/iceberg/io/TestDVWriters.java

```diff
@@ -326,6 +348,14 @@ private DeleteFile writePositionDeletes(
     return writer.toDeleteFile();
   }
 
+  private void assertNoPuffinFilesInDataDir() {
+    File dir = new File(table.location(), "data");
+    if (dir.exists()) {
```

Comment:
Does this condition become true in the current test? How about changing to `assertThat(dir).doesNotExist()` if the following code isn't reachable.